core/src/main/scala/spire/math/Real.scala

```diff
@@ -474,7 +476,19 @@ object Real extends RealInstances {
 
   lazy val sqrt1By2 = Real.two.reciprocal.sqrt
 
+  def accumulate(total: SafeLong, xs: LazyList[SafeLong], cs: LazyList[Rational]): SafeLong = {
+    (xs, cs) match {
+      case (_, Seq()) => total
+      case (Seq(), _) => sys.error("nooooo")
+      case (x #:: xs, c #:: cs) =>
+        val t = roundUp(c * Rational(x))
+        if (t == 0) total else accumulate(total + t, xs, cs)
+    }
+  }
+
+  @deprecated("prefer LazyList instead", "0.17.0")
   def accumulate(total: SafeLong, xs: Stream[SafeLong], cs: Stream[Rational]): SafeLong = {
+    import scala.#::
```

Comment:
Why is this import here?

Reply:
so... there is an `object #::` in the `scala` package object containing `unapply` for each of `LazyList` and `Stream` that forward to their respective extractors. unfortunately, in 2.12-, it's just an alias for `Stream`'s extractor, since `LazyList` didn't exist. Because of this, I had to import `LazyList.#::` at the top of the file to make *its* extractor work. That import ends up shadowing `scala.#::`, making `Stream`'s extractor no longer work, so I had to add this import. I wanted `LazyList` to be the "natural" one, but we could also remove this import and move the `LazyList.#::` import to a localised scope where it doesn't affect the `accumulate` overload for `Stream`